File: objetos/Planta.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package objetos;

import java.util.List;
import objetos.Tratamento;

/**
 *
 * @author anacs
 */
public abstract class Planta extends SerVivo {
    private String tipoSolo;
    private String cicloVida;
    private SistemaIrrigacao si;
    private boolean adubacao=false;


    
    @Override
    public void manejar() {
        System.out.println("Realizando manejo geral de planta: " + nome );
         for (Tratamento t : tratamentos) {
            System.out.println("- Data: " + t.getData() + ", Tipo: " + t.getTipo() + ", Produto: " + t.getProduto());
        }
        System.out.println();
    }
    
    public void verificaReprodutividade(){
        if("adulto".equals(cicloVida)){
            System.out.println("Planta " + nome + " está em fase reprodutiva.");
        } else {
            System.out.println("Planta " + nome + " não está em fase reprodutiva.");
        }
    }

    public String getTipoSolo() {
        return tipoSolo;
    }

    public void setTipoSolo(String tipoSolo) {
        this.tipoSolo = tipoSolo;
    }

    public String getCicloVida() {
        return cicloVida;
    }

    public void setCicloVida(String cicloVida) {
        this.cicloVida = cicloVida;
    }
    
    public void adubar(){
        adubacao=true;
    }
    public boolean isHidratada() {
        return si.isLigado();
    }
    public boolean isAdubado() {
        return adubacao;
    }

}

File: python/planta.py
from abc import abstractmethod
from ser_vivo import SerVivo

class Planta(SerVivo):
    def __init__(self):
        super().__init__()
        self.tipo_solo = None
        self.ciclo_vida = None
        self.si = None
        self.adubacao = False

    def manejar(self):
        print(f"Realizando manejo geral de planta: {self.nome}")
        for t in self.tratamentos:
            print(f"- Data: {t.get_data()}, Tipo: {t.get_tipo()}, Produto: {t.get_produto()}")
        print()

    def verifica_reprodutividade(self):
        if self.ciclo_vida == "adulto":
            print(f"Planta {self.nome} está em fase reprodutiva.")
        else:
            print(f"Planta {self.nome} não está em fase reprodutiva.")

    def get_tipo_solo(self):
        return self.tipo_solo

    def set_tipo_solo(self, tipo_solo):
        self.tipo_solo = tipo_solo

    def get_ciclo_vida(self):
        return self.ciclo_vida

    def set_ciclo_vida(self, ciclo_vida):
        self.ciclo_vida = ciclo_vida

    def adubar(self):
        self.adubacao = True

    def is_hidratada(self):
        return self.si.is_ligado() if self.si else False

    def is_adubado(self):
        return self.adubacao